src/compile/data/stack.ts

```diff
@@ -179,10 +179,15 @@ export class StackNode extends DataFlowNode {
 
     out.add(this._stack.stackField);
 
-    this.getGroupbyFields().forEach(out.add);
-    this._stack.facetby.forEach(out.add);
-    const field = this._stack.sort.field as string;
-    isArray(field) ? field.forEach(out.add) : out.add(field);
+    this.getGroupbyFields().forEach(f => out.add(f));
+    this._stack.facetby.forEach(f => out.add(f));
+
+    const field = this._stack.sort.field;
+    if (isArray(field)) {
+      (field as [string]).forEach(f => out.add(f));
```

Comment:
Why do you need to force it to be a string array?

Reply:
If it's not forced as a string array I receive a type error because it could also be an array of `FieldRef` or `ExprRef` and out is a set of strings. Would it be better to not force the type and call `toString()` on `f` in the `forEach`?

Reply:
I thought we agreed in the meeting to change the types so that the field is always a string.